### PR TITLE
OCP - Tekton Pipeline for the installation of z/OS Cloud Broker and z/OS Connect Operators

### PR DESCRIPTION
**Pull request by midhun6989**
Added Ansible Tasks in `ocp-pipelines` Ansible Role to create Tekton Pipeline for the installation of z/OS Cloud Broker and z/OS Connect Operators.

### CODE REVIEW

File: ocp/ansible/playbooks/zoscb-install.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+---
+- name: Deploy z/OS Cloud Broker
+  hosts: localhost
+  gather_facts: false
+  become: false
+  collections:
+    - ibm.ibm_zos_core
+  vars:
+    ansible_python_interpreter: /usr/bin/python3
```
[the remark this answers is not in the document]
Comment (midhun6989) [reply ocp/ansible/playbooks/zoscb-install.yaml]: Modified

File: ocp/ansible/playbooks/zosconnect-install.yaml [hunk 2]
```diff
@@ -0,0 +1,16 @@
+---
+- name: Deploy z/OS Connect
+  hosts: localhost
+  gather_facts: false
+  become: false
+  collections:
+    - ibm.ibm_zos_core
+  vars:
+    ansible_python_interpreter: /usr/bin/python3
```
[the remark this answers is not in the document]
Comment (midhun6989) [reply ocp/ansible/playbooks/zosconnect-install.yaml]: Modified

File: ocp/ansible/roles/ocp-pipelines/files/pipeline-project.yaml [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+kind: Project
+apiVersion: "project.openshift.io/v1"
+metadata:
+  name: "ibm-zmodstack-pipelines"
+  annotations:
+    openshift.io/description: "Project containing the OpenShift Pipelines for IBM Z and Cloud Modernization Stack installation"
+    openshift.io/display-name: "OpenShift Pipelines"
+spec:
+  finalizers:
+    - kubernetes
```
[the remark this answers is not in the document]
Comment (midhun6989) [reply ocp/ansible/roles/ocp-pipelines/files/pipeline-project.yaml]: Modified

File: ocp/ansible/roles/ocp-pipelines/templates/zmodstack-install-pipeline.yaml.j2 [hunk 4]
```diff
@@ -0,0 +1,58 @@
+---
+apiVersion: tekton.dev/v1beta1
+kind: Pipeline
+metadata:
+  name: zmodstack-install-pipeline
+spec:
+  params:
+    - default: '{{ zoscb }}'
```
[the remark this answers is not in the document]
Comment (midhun6989) [reply ocp/ansible/roles/ocp-pipelines/templates/zmodstack-install-pipeline.yaml.j2:8]: I can demonstrate a pipeline created and its execution with the user inputs in ARM / CFN for `zoscb` set to `true` and `zosconnect` set to `false`.

File: ocp/ansible/roles/ocp-pipelines/tasks/main.yml [hunk 5]
```diff
@@ -3,3 +3,55 @@
...
     src: "files/pipeline-subscription.yaml"
+
+- name: "Verify Red Hat OpenShift Pipelines Operator is installed successfully"
+  ansible.builtin.command: oc get tektonconfig config --kubeconfig ~/.kube/config
+  register: pipelines_status
+  changed_when: false
+  until:
+    - "'True' in pipelines_status.stdout"
+  retries: 30
+  delay: 30
```
[the remark this answers is not in the document]
Comment (ivandov) [reply ocp/ansible/roles/ocp-pipelines/tasks/main.yml:14]: The value `True` for ready when being fetched through the `oc` CLI, is still just a field in the CR's `status` block. If you use the `k8s` module, and look for the `status.conditions` list to contain the following:

```
    - lastTransitionTime: '2024-01-24T10:12:26Z'
      status: 'True'
      type: Ready
```

This is how the `oc` output shows the `True` that you see in the stdout under the `Ready` column.

[the remark this answers is not in the document]
Comment (ivandov) [reply ocp/ansible/roles/ocp-pipelines/tasks/main.yml:14]: To move this PR along, you can work on this tweak/enhancement in another PR.

**Review by ivandov — DISMISSED** [review]
LGTM - we can open issues for the minor enhancements we mentioned in conversations above.

**Review by ivandov — APPROVED** [review]
A little worried about the hardcoded index of `status.conditions`. In the future new conditions may be introduced to this API that may cause this logic to become non-functional.

**Pull request merged**